core/services/ocr2/delegate.go

```diff
@@ -473,6 +479,147 @@ func GetEVMEffectiveTransmitterID(jb *job.Job, chain evm.Chain, lggr logger.Suga
 	return spec.TransmitterID.String, nil
 }
 
+type coreConfig struct {
+	Command       string `json:"command"`
+	ProviderType  string `json:"provider_type"`
+	PluginName    string `json:"plugin_name"`
+	TelemetryType string `json:"telemetry_type"`
+}
+
+type PluginConfig struct {
+	CoreConfig   coreConfig `json:"core_config"`
+	PluginConfig json.RawMessage
+}
+
+func validateGenericPluginSpec(c coreConfig) error {
+	if c.Command == "" {
+		return errors.New("generic config invalid: must provide command path")
+	}
+
+	if c.PluginName == "" {
+		return errors.New("generic config invalid: must provide plugin name")
+	}
+
+	if c.TelemetryType == "" {
+		return errors.New("generic config invalid: must provide telemetry type")
+	}
+
+	return nil
+}
+
+type connProvider interface {
+	ClientConn() grpc.ClientConnInterface
+}
+
+func (d *Delegate) newServicesGenericPlugin(
+	ctx context.Context,
+	lggr logger.SugaredLogger,
+	jb job.Job,
+	bootstrapPeers []commontypes.BootstrapperLocator,
+	kb ocr2key.KeyBundle,
+	ocrDB *db,
+	lc ocrtypes.LocalConfig,
+	ocrLogger commontypes.Logger,
+	relayGetter RelayGetter,
+	registrar plugins.RegistrarConfig,
+	pipelineRunner pipeline.Runner,
+	endpointGen telemetry.MonitoringEndpointGenerator,
+) (srvs []job.ServiceCtx, err error) {
+	spec := jb.OCR2OracleSpec
+
+	p := PluginConfig{}
+	err = json.Unmarshal(spec.PluginConfig.Bytes(), &p)
+	if err != nil {
+		return nil, err
+	}
+	cconf := p.CoreConfig
+
+	err = validateGenericPluginSpec(cconf)
```

Comment:
Shouldn't this be in the `validateSpec` function in `core/services/ocr2/validate/validate.go` ?

Reply:
Uhm yes! It should be -- will change